lib/mainGame.dart

```diff
@@ -216,18 +216,50 @@ class MainGame extends Forge2DGame with MultiTouchDragDetector {
     if (egoSpaceship != null) {
       cameraFollow(egoSpaceship, horizontal: 0, vertical: 0);
 
-      // Send Spaceship information to Server
-      launcher.serverHandler.sendDataToServer(
-        action: "updateSpaceship",
-        data: {
-          "position": [egoSpaceship.body.position.x, egoSpaceship.body.position.y],
-          "angle": egoSpaceship.radAngle,
-          "resources": egoSpaceship.resources,
-        },
-      );
+      /* Only send spaceship updates to server if we're alive. 
+      Otherwise, client listens for respawnTime updates.
+      Once respawnTime hits 0, client resets its position and rotation
+      to initial values and server is updated in next tick. */
+      if (!egoSpaceship.isSpectating) {
+        updateServer(
+          {
+            "position": [egoSpaceship.body.position.x, egoSpaceship.body.position.y],
+            "angle": egoSpaceship.radAngle,
+            "resources": egoSpaceship.resources,
+            "isSpectating": egoSpaceship.isSpectating,
+          },
+        );
+      } else {
+        if (egoSpaceship.respawnTime != null && int.parse(egoSpaceship.respawnTime) == 0) {
```

Comment:
respawnTime should be initiated as 0, that way we don't have to do the null check.